web-local/src/lib/components/workspace/core/WorkspaceContainer.svelte

```diff
@@ -0,0 +1,72 @@
+<script lang="ts">
+  import {
+    SIDE_PAD,
+    SURFACE_SLIDE_DURATION,
+    SURFACE_SLIDE_EASING,
+  } from "@rilldata/web-local/lib/application-config";
+  import { localStorageStore } from "@rilldata/web-local/lib/components/stores/local-storage";
+  import { getContext, setContext } from "svelte";
+  import { tweened } from "svelte/motion";
+
+  import type { LayoutElement } from "@rilldata/web-local/lib/types";
+  import type { Writable } from "svelte/store";
+  import Inspector from "./Inspector.svelte";
+
+  export let assetID;
+  export let inspector = true;
+  export let bgClass = "bg-gray-100";
+
+  const inspectorLayout = localStorageStore<LayoutElement>(
+    { value: inspector ? 400 : 0, visible: true },
+    assetID
+  );
+  const inspectorWidth = tweened(
+    inspector ? $inspectorLayout?.value || 400 : 0,
+    {
+      duration: 50,
+    }
+  );
+  inspectorLayout.subscribe((state) => {
+    inspectorWidth.set(state.value);
+  });
+
+  export const visibilityTween = tweened($inspectorLayout?.visible ? 1 : 0, {
+    duration: SURFACE_SLIDE_DURATION,
+    easing: SURFACE_SLIDE_EASING,
+  });
+
+  setContext("rill:app:inspector-layout", inspectorLayout);
+  setContext("rill:app:inspector-width-tween", inspectorWidth);
+  setContext("rill:app:inspector-visibility-tween", visibilityTween);
+
+  const navigationWidth = getContext(
+    "rill:app:navigation-width-tween"
+  ) as Writable<number>;
+  const navVisibilityTween = getContext(
+    "rill:app:navigation-visibility-tween"
+  ) as Writable<number>;
+
+  let hasNoError = 1;
+  let hasInspector = true;
```

Comment:
Same here - it doesn't look like either `hasNoError` or `hasInspector` gets manipulated anywhere. Remove?